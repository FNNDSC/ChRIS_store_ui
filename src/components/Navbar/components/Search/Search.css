.form-group.has-icon .form-control {
  /* display */
  padding-left: 24px;
}

.search-pf .has-icon .search-icon {
  /* display */
  height: 25px;
  width: 20px;

  /* font */
  line-height: 1.6em;
  text-align: center;
  font-size: 1.2em;
  
  /* position */
  position: absolute;
  left: 1px;
  top: 1px;
}