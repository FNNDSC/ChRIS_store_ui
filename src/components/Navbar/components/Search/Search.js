import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'patternfly-react';
import './Search.css';

// a helper function to get query params from page url
const getQueryParams = (search) => {
  const groups = search.slice(1).split('&');
  const splitGroups = groups.map(group => group.split('='));
  const queryParams = splitGroups.reduce((acc, group) => {
    const [key, value] = group;
    acc[key] = value;
    return acc;
  }, {});
  return queryParams;
};

const Search = ({ className, location }) => {
  const { search } = location;
  const { q: query } = getQueryParams(search);
  return (
    <form
      method="get"
      action="/plugins"
      autoComplete="off"
      className={`search-pf ${className}`}
    >
      <div className="form-group has-icon">
        <div className="search-pf-input-group">
          <div className="search-icon">
            <Icon name="search" />
          </div>
          <input
            type="search"
            id="search"
            className="form-control"
            name="q"
            placeholder="Search plugins"
            defaultValue={query}
            autoComplete="off"
          />
        </div>
      </div>
    </form>
  );
};

Search.propTypes = {
  className: PropTypes.string,
  location: PropTypes.shape({
    search: PropTypes.string,
  }),
};
Search.defaultProps = {
  className: '',
  location: {
    search: '',
  },
};

export default Search;
