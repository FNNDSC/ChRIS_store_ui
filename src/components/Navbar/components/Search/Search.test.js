import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

/* define props based on an object template */
const propsToMatch = (wrapper, template) => Object.keys(template)
  .every(key => wrapper.prop(key) === template[key]);

describe('Search', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Search />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render form with class search-pf', () => {
    expect(wrapper.find('form.search-pf')).toHaveLength(1);
  });

  it('should render search-pf form with required props', () => {
    const form = wrapper.find('form.search-pf');
    const template = {
      method: 'get',
      action: '/plugins',
      autoComplete: 'off',
    };

    expect(propsToMatch(form, template)).toBeTruthy();
  });

  it('should render form-group div inside form', () => {
    expect(wrapper
      .find('form')
      .find('div.form-group'))
      .toHaveLength(1);
  });

  it('should render search-pf-input-group div inside form-group', () => {
    expect(wrapper
      .find('div.form-group')
      .find('div.search-pf-input-group'))
      .toHaveLength(1);
  });

  it('should render search-icon div inside search-pf-input-group', () => {
    expect(wrapper
      .find('div.search-pf-input-group')
      .find('div.search-icon'))
      .toHaveLength(1);
  });

  it('should render Icon inside search-icon div', () => {
    expect(wrapper
      .find('div.search-icon')
      .find('Icon'))
      .toHaveLength(1);
  });

  it('should render form-control input inside search-pf-input-group', () => {
    expect(wrapper
      .find('div.search-pf-input-group')
      .find('input#search.form-control'))
      .toHaveLength(1);
  });

  it('should render form-control with required props', () => {
    const input = wrapper.find('input#search.form-control');
    const template = {
      type: 'search',
      name: 'q',
      placeholder: 'Search plugins',
    };

    expect(propsToMatch(input, template)).toBeTruthy();
  });

  /* ====== PROP BASED TESTS ====== */

  it('should render className prop value', () => {
    wrapper.setProps({ className: 'testClass' });
    expect(wrapper
      .find('form.search-pf')
      .hasClass('testClass'))
      .toBeTruthy();
  });

  it('should render input defaultValue based off location prop', () => {
    wrapper.setProps({
      location: {
        search: '?q=testQuery',
      },
    });
    const input = wrapper.find('input#search.form-control');
    const template = { defaultValue: 'testQuery' };

    expect(propsToMatch(input, template));
  });
});
