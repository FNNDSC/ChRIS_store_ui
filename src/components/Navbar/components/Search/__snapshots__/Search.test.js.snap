// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    className=""
    location={
        Object {
            "search": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "action": "/plugins",
      "autoComplete": "off",
      "children": <div
        className="form-group has-icon"
>
        <div
                className="search-pf-input-group"
        >
                <div
                        className="search-icon"
                >
                        <Icon
                                name="search"
                                type="fa"
                        />
                </div>
                <input
                        autoComplete="off"
                        className="form-control"
                        defaultValue={undefined}
                        id="search"
                        name="q"
                        placeholder="Search plugins"
                        type="search"
                />
        </div>
</div>,
      "className": "search-pf ",
      "method": "get",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="search-pf-input-group"
>
          <div
                    className="search-icon"
          >
                    <Icon
                              name="search"
                              type="fa"
                    />
          </div>
          <input
                    autoComplete="off"
                    className="form-control"
                    defaultValue={undefined}
                    id="search"
                    name="q"
                    placeholder="Search plugins"
                    type="search"
          />
</div>,
        "className": "form-group has-icon",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="search-icon"
>
              <Icon
                            name="search"
                            type="fa"
              />
</div>,
            <input
              autoComplete="off"
              className="form-control"
              defaultValue={undefined}
              id="search"
              name="q"
              placeholder="Search plugins"
              type="search"
/>,
          ],
          "className": "search-pf-input-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                name="search"
                type="fa"
/>,
              "className": "search-icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "search",
                "type": "fa",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoComplete": "off",
              "className": "form-control",
              "defaultValue": undefined,
              "id": "search",
              "name": "q",
              "placeholder": "Search plugins",
              "type": "search",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "/plugins",
        "autoComplete": "off",
        "children": <div
          className="form-group has-icon"
>
          <div
                    className="search-pf-input-group"
          >
                    <div
                              className="search-icon"
                    >
                              <Icon
                                        name="search"
                                        type="fa"
                              />
                    </div>
                    <input
                              autoComplete="off"
                              className="form-control"
                              defaultValue={undefined}
                              id="search"
                              name="q"
                              placeholder="Search plugins"
                              type="search"
                    />
          </div>
</div>,
        "className": "search-pf ",
        "method": "get",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="search-pf-input-group"
>
            <div
                        className="search-icon"
            >
                        <Icon
                                    name="search"
                                    type="fa"
                        />
            </div>
            <input
                        autoComplete="off"
                        className="form-control"
                        defaultValue={undefined}
                        id="search"
                        name="q"
                        placeholder="Search plugins"
                        type="search"
            />
</div>,
          "className": "form-group has-icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="search-icon"
>
                <Icon
                                name="search"
                                type="fa"
                />
</div>,
              <input
                autoComplete="off"
                className="form-control"
                defaultValue={undefined}
                id="search"
                name="q"
                placeholder="Search plugins"
                type="search"
/>,
            ],
            "className": "search-pf-input-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  name="search"
                  type="fa"
/>,
                "className": "search-icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "name": "search",
                  "type": "fa",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "off",
                "className": "form-control",
                "defaultValue": undefined,
                "id": "search",
                "name": "q",
                "placeholder": "Search plugins",
                "type": "search",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
